--- /tmp/mdlm_orig/modeling_mdlm_2.py	2026-09-01 16:02:53.898410114 +0000
+++ modeling_mdlm_2.py	2026-09-01 15:57:48.222328111 +0000
@@ -5,8 +5,8 @@
 import typing
 
 import einops
-import flash_attn
-import flash_attn.layers.rotary
+# import flash_attn
+# import flash_attn.layers.rotary
 import huggingface_hub
 import omegaconf
 import torch
@@ -117,9 +117,7 @@
 
 
 def apply_rotary_pos_emb(qkv, cos, sin):
-  cos = cos[0,:,0,0,:cos.shape[-1]//2]
-  sin = sin[0,:,0,0,:sin.shape[-1]//2]
-  return flash_attn.layers.rotary.apply_rotary_emb_qkv_(qkv, cos, sin)
+  return qkv * cos + rotate_half(qkv) * sin
 
 
 # function overload
@@ -427,6 +425,7 @@
     config: MDLMConfig):
     super().__init__(config)
     self.backbone = DITBackbone(config)
+    self.post_init()
 
   def forward(
       self,
